### PR TITLE
[observability pipelines worker] 1.2.0-rc.0 release

### PR DESCRIPTION
**Pull request by 001wwang**
#### What this PR does / why we need it:

Helm Chart changes required by OPW 1.2.0

#### Which issue this PR fixes
* ~closes [OPB-704](https://datadoghq.atlassian.net/browse/OPB-704)~
* closes [OPB-751](https://datadoghq.atlassian.net/browse/OPB-751) 

#### Special notes for your reviewer:

Drafted pending questions and OPW 1.2.0 release

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

[OPB-704]: https://datadoghq.atlassian.net/browse/OPB-704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OPB-751]: https://datadoghq.atlassian.net/browse/OPB-751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: charts/observability-pipelines-worker/ci/all-values.yaml
Comment (001wwang) [inline charts/observability-pipelines-worker/ci/all-values.yaml:3]: Note, key validation work is still blocked on the OPW side -- if changes occur, this will need updating.

File: charts/observability-pipelines-worker/ci/dupe-ports-values.yaml
Comment (001wwang) [inline charts/observability-pipelines-worker/ci/dupe-ports-values.yaml:5]: @spencergilbert let me know if this is correct/if there's a way to avoid this duplicated override.

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (spencergilbert) [inline charts/observability-pipelines-worker/templates/configmap.yaml:13]: The filename has become `pipelines.yaml`, do you still want the user to configure it's contents via a key `config`?

File: charts/observability-pipelines-worker/Chart.yaml
Comment (spencergilbert) [inline charts/observability-pipelines-worker/Chart.yaml:16]: Can technically leave this as-is if you want to change things in one less place. The Datadog Agent also just puts the major version here (it's not functionally used, just documentation)

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (001wwang) [inline charts/observability-pipelines-worker/templates/configmap.yaml:11]: I'd like to get more familiar with the wiring up of the config. Is the following correct?

- Users must provide a config value
- The config is turned into a `ConfigMap` (i.e. here)
- The `ConfigMap` is made available in a volume at `/etc/observability-pipelines-worker/`
- OPW can then read from `/etc/observability-pipelines-worker/pipelines.yaml`

And if users already have their own config file at a different location, are they meant to override the `args` to use it?

File: charts/observability-pipelines-worker/ci/dupe-ports-values.yaml
Comment (spencergilbert) [reply charts/observability-pipelines-worker/ci/dupe-ports-values.yaml:5]: I _think_ this is right, without needing to change the shared CI workflow. We can also look to remove some of the cases, they may not all be needed with the more streamlined OPW experience.

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/templates/configmap.yaml:13]: Yes, good timing, I was about to comment here. I think `config` is too general, especially with bootstrap involved. Is there a good way to refactor this other than find/replace? I think `pipelinesConfig` (open to suggestions) may be better.

File: charts/observability-pipelines-worker/Chart.yaml
Comment (spencergilbert) [reply charts/observability-pipelines-worker/Chart.yaml:16]: Less updating/they're more strict on semver I'd assume. Vector used this `appVersion` to populate the `image.tag` - but I followed the existing DD chart's pattern so the setup would be more familiar for anyone reviewing PRs from the org.

**Comment by 001wwang** [comment]
@spencergilbert One large remaining piece is handling `bootstrap.yaml` (contents pending https://github.com/DataDog/observability-pipelines-worker/pull/66). We will be adding a default file to the image itself, but what does wiring (if any) look like for the Helm Chart (i.e. making it easy for users to set the environment variables, etc.)?

File: charts/observability-pipelines-worker/Chart.yaml
Comment (spencergilbert) [reply charts/observability-pipelines-worker/Chart.yaml:16]: TL;DR is y'alls choice between specificity/accuracy and desire to change one less line per release 😆

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (spencergilbert) [reply charts/observability-pipelines-worker/templates/configmap.yaml:13]: Yeah, that name seems fine to me? I'm terrible with names though. Find and replace should do it.

We'll probably need to make a couple more changes to account for the `bootstrap.yaml` however.

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (spencergilbert) [reply charts/observability-pipelines-worker/templates/configmap.yaml:11]: 1. User provides contents to the `config` key in the `values.yaml`.
2. `configmap.yaml` will take those contents try to parse them as YAML.
3. This [volume](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L117-L121) is created using the ConfigMap from 2.
4. This [volumeMount](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L84-L86) places the contents of the volume at `/etc/observability-pipelines-worker/` - the configuration we provided creates each key under `data` as a file there.
5. The provided file is accessible at `/etc/observability-pipelines-worker/pipelines.yaml` during runtime.

At this point the only file path the config will be placed is the above, and the only method of providing a config is through the `config` key. Do we think users actually need this supported? Off the top of my head I can't think of anything that would require that.

**Comment by spencergilbert** [comment]
> @spencergilbert One large remaining piece is handling `bootstrap.yaml` (contents pending [DataDog/observability-pipelines-worker#66](https://github.com/DataDog/observability-pipelines-worker/pull/66)). We will be adding a default file to the image itself, but what does wiring (if any) look like for the Helm Chart (i.e. making it easy for users to set the environment variables, etc.)?

We have a couple of environment variables that can be specified directly [here](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L45-L56). We could provide a similar `config|pipelinesConfig` key to override the file, but if all of the options can currently be set via variables I feel like that's a more straightforward.

File: charts/observability-pipelines-worker/Chart.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/Chart.yaml:16]: Ok, in the spirit of following existing practices and since I don't have much expertise here, I'm going to leave this as is.

**Comment by 001wwang** [comment]
> We have a couple of environment variables that can be specified directly [here](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L45-L56). We could provide a similar config|pipelinesConfig key to override the file, but if all of the options can currently be set via variables I feel like that's a more straightforward.

Yes, environment variables is fine; we'll default bootstrap pre-filled with env vars. As mentioned elsewhere, I'm starting to see we should encourage users to interact with this setup via `values`.

Other than what's already here, we're missing `DD_ENDPOINT` and `DD_OP_DATA_DIR`. The `data_dir` volume that's created right now seems wired to the pipeline config value. I think we need to change this -- maybe have both since it's possible they could configure one in bootstrap config and another in the pipeline config (not sure why they would, but it's possible)?

**Comment by 001wwang** [comment]
I likely won't have a chance to update this personally before my out-of-office, so leaving a status update for now. This is awaiting the following changes at least

- Rename `config` in `values` to `pipelineConfig` throughout
- Ensure all environment variables in the default `bootstrap.yaml` can be easily set (`DD_OP_DATA_DIR`, `DD_ENDPOINT` remaining)
- Any updates related to https://datadoghq.atlassian.net/browse/OPB-750

File: charts/observability-pipelines-worker/ci/all-values.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/ci/all-values.yaml:3]: Changes have been merged on the OPW side.

File: charts/observability-pipelines-worker/templates/configmap.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/templates/configmap.yaml]: I've updated the naming to `pipelineConfig` in [31a5536](https://github.com/DataDog/helm-charts/pull/952/commits/31a55360b1dcf5d876ff1087c50c54bcca9950e1) and beyond.

**Comment by 001wwang** [comment]
Hey @spencergilbert, I've brought this up to date with the following changes
- Updating `config` to `pipelineConfig`
- Adding values to set `endpoint` and `data_dir` for the bootstrap file
- Updating test configs, README, etc.

Let me know how that all looks. 

Thinking more, I'm wondering if we should offer a way to override the default bootstrap and allow a user to pass one optionally via a `bootstrapConfig` value. Do you think this would be a valid use case or would setting the environment variables that feed into the default bootstrap be sufficient?

File: charts/observability-pipelines-worker/templates/_pod.tpl
Comment (spencergilbert) [inline charts/observability-pipelines-worker/templates/_pod.tpl:87]: Curious how OPW handles things if a `data_dir` is specified in both the pipelines config and the bootstrap/opw side.

File: charts/observability-pipelines-worker/CHANGELOG.md
Comment (spencergilbert) [inline charts/observability-pipelines-worker/CHANGELOG.md:7]:
```suggestion
* Add `datadog.endpoint` and `datadog.data_dir` values
```

**Review by spencergilbert — COMMENTED** [review]
> Let me know how that all looks.

Changes look good, small nits for technically correctness.
 
> Thinking more, I'm wondering if we should offer a way to override the default bootstrap and allow a user to pass one optionally via a `bootstrapConfig` value. Do you think this would be a valid use case or would setting the environment variables that feed into the default bootstrap be sufficient?

As I understand it, outside of Kubernetes, the user would override the values _primarily_ through the environment variables. I'd suggest leaving it just as environment variables unless someone requests it particularly - I can't think of anything that a file would solve that the env vars don't.

Keep it simple until it's needed, imho

File: charts/observability-pipelines-worker/templates/_pod.tpl
Comment (001wwang) [reply charts/observability-pipelines-worker/templates/_pod.tpl:87]: As of https://github.com/DataDog/observability-pipelines-worker/pull/81, if `data_dir` is set in the pipeline config, we'll raise a validation error and OPW will fail to start e.g.
```
Configuration is invalid.
-------------------------
x Pipeline configuration must not contain data_dir setting. This should be set in the bootstrap configuration.
```

Comment (spencergilbert) [reply charts/observability-pipelines-worker/templates/_pod.tpl:87]: Ah, that sounds familiar - just wanted to make sure that's covered.

**Comment by spencergilbert** [comment]
RE: default `pipeline.yaml`, I think you'll just need to add a conditional around the `config` volume and volumeMount to not render those if the `pipelineConfig` is empty. Happy to help you get that done if you need to pair.

**Comment by 001wwang** [comment]
For posterity, @spencergilbert and I met to discuss updates required for using a default `pipeline.yaml`. I've pushed a few updates to handle basic cases where `.Values.pipelineConfig` is empty, and @spencergilbert will look into changes related to service ports.

**Comment by 001wwang** [comment]
Noting I'll make another update to take https://datadoghq.atlassian.net/browse/OPB-780 into account. In short, I think we'd like to change
- `configKey` to `pipelineID`
- `endpoint` to `dd_url`

Note, `DD_OP_CONFIG_KEY` and `DD_ENDPOINT` are still supported aliases for the newer `DD_OP_PIPELINE_ID` and `DD_URL`. I think we can similarly leave `configKey` and `endpoint` available here if backwards compatibility is needed.

Since we've already been going with env var based bootstrap configuration in this PR, I don't think anything else needs updating.

cc: @spencergilbert for context.

File: charts/observability-pipelines-worker/README.md
Comment (001wwang) [inline charts/observability-pipelines-worker/README.md:27]: @spencergilbert I made some updates in [48707e1](https://github.com/DataDog/helm-charts/pull/952/commits/48707e1d02ddbb51ac958161105f9a6bd2ad0280) and [30d5b0a](https://github.com/DataDog/helm-charts/pull/952/commits/30d5b0ad0c8c283bee223fda61379c252c671cab). Let me know how that looks.

I'm curious if we absolutely need secrets management for the config key/pipeline ID? It's essentially just a UUID identifier for a particular pipeline, so I'm not sure it's considered sensitive data. This is mostly for my own understanding, I think we're fine to leave as is for now.

**Comment by 001wwang** [comment]
Another heads up: we've decided to remove the default pipeline configuration, so I'll walk back some of my other changes and leave `pipelineConfig` as required.

File: charts/observability-pipelines-worker/README.md
Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md:27]: If it's just a UUID, it may not need to be a Secret. Could anything funky happen if an attacker reused the same ID?

I think I was under the impression that it was suppose to be a confidential value when I wrote this, if not it could just as easily be a ConfigMap.

File: charts/observability-pipelines-worker/README.md
Comment (spencergilbert) [inline charts/observability-pipelines-worker/README.md:28]:
```suggestion
manually created Secrets by setting the `datadog.apiKeyExistingSecret` and/or `datadog.pipelineIdExistingSecret` values
```

If we're being super proper/pedantic it should be `Id`.

File: charts/observability-pipelines-worker/values.yaml
Comment (spencergilbert) [inline charts/observability-pipelines-worker/values.yaml:40]:
```suggestion
  url: ""
```

I think this name looks better especially when written with the parent: `datadog.ddURL` vs `datadog.url`.

File: charts/observability-pipelines-worker/values.yaml
Comment (spencergilbert) [inline charts/observability-pipelines-worker/values.yaml:28]:
```suggestion
  pipelineId:  # <DD_OP_PIPELINE_ID>
```

Nit/pedantic

File: charts/observability-pipelines-worker/values.yaml
Comment (spencergilbert) [inline charts/observability-pipelines-worker/values.yaml:31]:
```suggestion
  pipelineIdExistingSecret: ""
```

Nit/pedantic

File: charts/observability-pipelines-worker/templates/_helpers.tpl
Comment (spencergilbert) [inline charts/observability-pipelines-worker/templates/_helpers.tpl:54]:
```suggestion
{{- define "opw.pipelineIdSecretName" -}}
{{- $fullName := printf "%s-pipeline-id" (include "opw.fullname" .) -}}
```

Nit/pedantic

File: charts/observability-pipelines-worker/templates/_helpers.tpl
Comment (spencergilbert) [inline charts/observability-pipelines-worker/templates/_helpers.tpl:51]: This replaces the `opw.configKeySecretName` partial above, correct? We should remove it if it's now unused.

File: charts/observability-pipelines-worker/CHANGELOG.md
Comment (spencergilbert) [inline charts/observability-pipelines-worker/CHANGELOG.md:7]:
```suggestion
* Add `datadog.pipelineId` value to replace `datadog.configKey`
```

Or with the current styling of `pipelineID`, if you still prefer that. Probably also worth noting in the change log that `configKey` should still works for backwards compatibility?

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: charts/observability-pipelines-worker/README.md
Comment (001wwang) [reply charts/observability-pipelines-worker/README.md:28]: 😢 Alas, I flip-flopped on this and landed on ID. Let's go for super proper 👍

Comment (001wwang) [reply charts/observability-pipelines-worker/README.md:28]: I'll bundle the renaming's into a single commit.

Comment (001wwang) [reply charts/observability-pipelines-worker/README.md]: Adjusted in [7a5e07c](https://github.com/DataDog/helm-charts/pull/952/commits/7a5e07c5b852e9393a9f49a75395fe08597c52dd)

File: charts/observability-pipelines-worker/values.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/values.yaml]: Adjusted in [7a5e07c](https://github.com/DataDog/helm-charts/pull/952/commits/7a5e07c5b852e9393a9f49a75395fe08597c52dd)

File: charts/observability-pipelines-worker/values.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/values.yaml]: Adjusted in [7a5e07c](https://github.com/DataDog/helm-charts/pull/952/commits/7a5e07c5b852e9393a9f49a75395fe08597c52dd)

File: charts/observability-pipelines-worker/templates/_helpers.tpl
Comment (001wwang) [reply charts/observability-pipelines-worker/templates/_helpers.tpl:51]: I think this is still necessary for backwards compatibility unless I'm misunderstanding something here.

File: charts/observability-pipelines-worker/CHANGELOG.md
Comment (001wwang) [reply charts/observability-pipelines-worker/CHANGELOG.md]: Sounds good, adjusted in [7a5e07c](https://github.com/DataDog/helm-charts/pull/952/commits/7a5e07c5b852e9393a9f49a75395fe08597c52dd)

File: charts/observability-pipelines-worker/values.yaml
Comment (001wwang) [reply charts/observability-pipelines-worker/values.yaml]: Yep, sounds good, updated in [cc13d58](https://github.com/DataDog/helm-charts/pull/952/commits/cc13d58f845d5c24df6b474fbad2d4925203a928)

File: charts/observability-pipelines-worker/README.md
Comment (001wwang) [reply charts/observability-pipelines-worker/README.md:27]: > Could anything funky happen if an attacker reused the same ID?

Off the top of my head, I don't think anything malicious can be done with solely the pipeline ID. 

> I think I was under the impression that it was suppose to be a confidential value when I wrote this, if not it could just as easily be a ConfigMap.

Right, I'm wondering if we could treat it like the other values under `datadog`. Perhaps we could simplify this in a follow up? It doesn't seem particularly urgent.

File: charts/observability-pipelines-worker/templates/_helpers.tpl
Comment (spencergilbert) [reply charts/observability-pipelines-worker/templates/_helpers.tpl:51]: Oops, yeah - I temporarily lost my mind and forgot about the backward compatibility here.

File: charts/observability-pipelines-worker/README.md
Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md:27]: Ah, yeah if it's determined there's not an issue exposing it we could transition it to similar to the `datadog.site` or `datadog.url` options.

I'd want to doublecheck the assumption before making that happen, for sure.

**Review by spencergilbert — APPROVED** [review]
The changes LGTM at this point

File: charts/observability-pipelines-worker/README.md
Comment (leebenson) [reply charts/observability-pipelines-worker/README.md:27]: > Off the top of my head, I don't think anything malicious can be done with solely the pipeline ID.

Just a drive-by...

Pipeline ID isn't intended to be sensitive. It's really just a resource ID, embedded in URLs and made public in UI without regard to obfuscation. Doesn't need to be a secret.

Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md:27]: Definitely sounds like something to change then, apologies for my misunderstanding!

Comment (001wwang) [reply charts/observability-pipelines-worker/README.md:27]: @spencergilbert, for simplicity's sake, how do we feel about
- Treat `pipelineId` similar to `datadog.___` options
- Remove the `configKey` secret entirely

as part of this release?

Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md:27]: No objections for the first. For the second, that may be considered a breaking change if users had been using the `existingConfigKey` options.

It would complicate the code a here a bit, but we could do `datadog.pipelinesId` or `datadog.configKey` goes directly in the Pod's `env` block (both creating the _new_ env var name), and `datadog.configKeyExistingSecret` still using a referenced Secret to populate it. Marking the two `configKey` options as deprecated and dropping next release?

Comment (001wwang) [reply charts/observability-pipelines-worker/README.md:27]: > No objections for the first.

Ok, I'll make some minor updates then for `pipelineId`.

> It would complicate the code a here a bit, but we could do datadog.pipelinesId or datadog.configKey goes directly in the Pod's env block (both creating the new env var name), and datadog.configKeyExistingSecret still using a referenced Secret to populate it. Marking the two configKey options as deprecated and dropping next release?

Sure, this seems reasonable. Would you be able to make the updates here?

To make sure I'm on the right page, we're planning to set `DD_OP_PIPELINE_ID` in the environment in order of precedence from 1) `datadog.pipelineId` 2) `datadog.configKey` 3) `datadog.configKeyExistingSecret`? And we can get rid of some current plumbing that creates a secret from `datadog.configKey`?

Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md]: Yeah, I can make some time tonight to do that. Your understanding sounds right, though I'll double check that there's no issues with `datadog.configKey` no longer creating a Secret - between versions.

Comment (spencergilbert) [reply charts/observability-pipelines-worker/README.md]: Pushed some changes in https://github.com/DataDog/helm-charts/pull/952/commits/34e8658df19b73bed271faa39d5b2c7036313c65, I left the secret as-is since I don't think Helm will natively remove the orphaned resources.

**Comment by sghall** [comment]
@001wwang just wanted to note some feedback we received from @jszwedko in the [Feature Fest doc](https://docs.google.com/document/d/1aC5PNYvx0dc1xrdxTQyHUqNm6URgvnPAa8d0BifUdcE/edit#).  

> …when attempting to use datadog.pipelineId…it gives a warning that configKey isn’t set

Just want to make sure we have addressed that before merging this one 👍

**Comment by sghall** [comment]
Also noting we need align on enabling RC.  Instructions PR up over here https://github.com/DataDog/web-ui/pull/89053

Instructions now look like...
Screenshot 2023-05-02 at 10 11 03 AM

**Comment by 001wwang** [comment]
Hey @spencergilbert, I pushed a few changes, fixing some things mentioned in :
- [efb3e02](https://github.com/DataDog/helm-charts/pull/952/commits/efb3e02b7ca4587252b1f65bfd6f932d8206ce53) Avoids the error message referencing `configKey` when `pipelineId` is set.
- [143ed0f](https://github.com/DataDog/helm-charts/pull/952/commits/143ed0f4aa80625d0b6b1abf08941dfaa26a6bd6), [9fc2d4a](https://github.com/DataDog/helm-charts/pull/952/commits/9fc2d4a1c0079168ddfe687ed1ce6608ac77f8d8) Add and wire up `datadog.remoteConfigurationEnabled`
- [aba6253](https://github.com/DataDog/helm-charts/pull/952/commits/aba6253344a5bca17ca3cc2a1c895d113c3c0089) Remove the explicit `--pipeline-config` in default args -- OPW should find the default mounted to `/etc/observability-pipelines-worker/pipeline.yaml` automatically. This avoids an error if remote config is enabled wherein `--pipeline-config` is then not allowed.
- [2ad4b3c](https://github.com/DataDog/helm-charts/pull/952/commits/2ad4b3c5cf20146814778afa5def91f78d3ca6d5) Updating this to be a release candidate rather than official 1.2.0.

Let me know how that looks. A few things specifically I'd appreciate a double check on:
- Conditionally rendering the `configmap` based on whether remote config is enabled
- Any other places `configKey` might accidentally be an issue despite `pipelineId` being set
- Any other updates necessary for release candidate
- Any updates we should make to any docs in this repo

File: charts/observability-pipelines-worker/CHANGELOG.md
Comment (spencergilbert) [inline charts/observability-pipelines-worker/CHANGELOG.md:3]:
```suggestion
# 1.2.0-rc.0
```

File: charts/observability-pipelines-worker/README.md.gotmpl
Comment (spencergilbert) [inline charts/observability-pipelines-worker/README.md.gotmpl:56]:
```suggestion
```

Since you're moving away from the pipelineId/configKey from being a secret you can probably drop this?

File: charts/observability-pipelines-worker/templates/_pod.tpl
Comment (spencergilbert) [inline charts/observability-pipelines-worker/templates/_pod.tpl:53]:
```suggestion
      {{- with .Values.datadog.pipelineId }}
        value: {{ . | quote }}
      {{- else }}
```

Formatting nit, matches the indention on the lines below.

**Review by spencergilbert — CHANGES_REQUESTED** [review]
I'd consider adding a new `ci/` file to test the behavior of the new RC bits since that's a pretty critical user path going forward.

**Review by spencergilbert — CHANGES_REQUESTED** [review]

**Comment by 001wwang** [comment]
> I'd consider adding a new ci/ file to test the behavior of the new RC bits since that's a pretty critical user path going forward.

Yea, sounds good -- I think this should just involve a test config with `datadog.remoteConfigurationEnabled = true` and no pipeline config. 

On this note though, what exactly are the tests in `ci/` asserting? Is it just that the chart is installed successfully and all the k8s pieces are running?

**Comment by spencergilbert** [comment]
> > I'd consider adding a new ci/ file to test the behavior of the new RC bits since that's a pretty critical user path going forward.
> 
> Yea, sounds good -- I think this should just involve a test config with `datadog.remoteConfigurationEnabled = true` and no pipeline config.
> 
> On this note though, what exactly are the tests in `ci/` asserting? Is it just that the chart is installed successfully and all the k8s pieces are running?

Yeah, just setting that flag to true. It's not _truly_ end-to-end, but it'll attempt to install/upgrade the chart using those values so if we have any invalid resources or say the pod still trying to mount the non-existing configmap it'll error. If the image itself crashes due to error (for whatever reason), that'll also fail the CI job.

**Review by spencergilbert — APPROVED** [review]

**Comment by 001wwang** [comment]
Note for posterity, I squashed all commits to satisfy a repo requirement for signed commits (my previous commits were not signed).

**Pull request merged**
